### PR TITLE
feat: `make install-dependencies`

### LINKED ISSUE #30: Complete the `make setup` command for initialising the IBC client and channel
This supersedes: https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/8 and https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/7

We need to ensure that `make setup` does the following:

- [ ] Deploys the four IBC solidity contracts on the reth rollup
- [ ] Initialise the groth16 client on simapp that correctly references the state of the reth rollup
- [ ] Setup channels on both the reth rollup and simapp
- [ ] Register counterparth on each of the reth and simapp channels

This should prepare users to then call `make transfer`

### PR DESCRIPTION
**Pull request by ninabarbakadze**
## Overview

fixes #30

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,13 +22,21 @@ help: Makefile
 	@sed -n 's/^##//p' $< | column -t -s ':' |  sed -e 's/^/ /'
 .PHONY: help
 
+setup-solidity-ibc-eureka:
```
Comment (cmwaters) [inline Makefile:25]: I would prefer to call it something like install-dependencies

File: testing/demo/pkg/setup-env/setup-env.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
```
Comment (ninabarbakadze) [inline testing/demo/pkg/setup-env/setup-env.go:10]: do we still want to add a copy of .env in the root? I think it's good to separate different envs. @cmwaters @rootulp

File: Makefile (hunk 1 above)
Comment (ninabarbakadze) [reply Makefile]: https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/38/commits/a709328586612f3f74aea241a6ff99997aa0a0c1

File: testing/demo/pkg/setup-env/setup-env.go (hunk 2 above)
Comment (rootulp) [reply testing/demo/pkg/setup-env/setup-env.go:10]: Are there environment variables that we need set by the repo root that aren't present in the `solidity-ibc-eureka/.env`?

Comment (ninabarbakadze) [reply testing/demo/pkg/setup-env/setup-env.go:10]: yeah @cmwaters mentioned that he'll need env for his service

Comment (rootulp) [reply testing/demo/pkg/setup-env/setup-env.go:10]: I'm not strongly opinionated on where where we store the .envs or how many .env files we have. 

I'm fine with one root level .env that defines defaults for all the env variables the repo uses.

**Review by cmwaters — APPROVED** [review]

**Pull request merged**